app/assets/javascripts/bank_accounts.js.erb

```diff
@@ -1,4 +1,43 @@
 $(function() {
+  <% environment.context_class.instance_eval { include Rails.application.routes.url_helpers} %>
+
+  $('#add-bank-account-manual-button').click(function() {
+    $("#add-bank-account-manual-button").hide();
+    $("#manual-bank-intro-text").hide();
+    $('#manual-bank-account-form').removeClass('hidden');
+  });
+
+  $('#add-bank-account-plaid-button').click(function() {
+    $("#bank-account-options").hide()
+    $("#bank-account-please-wait").show();
+    var linkHandler = Plaid.create({
+      selectAccount: true,
+      clientName: 'Epicodus',
+      product: ['auth'],
+      env: '<%= Rails.env.production? ? 'production' : 'sandbox' %>',
+      key: '<%= ENV['PLAID_PUBLIC_KEY'] %>',
+      onSuccess: function(public_token, metadata) {
+        $.ajax({
+          method: "POST",
+          dataType: "script",
+          url: "<%= url_for(bank_accounts_path) %>",
+          data: { public_token: public_token, account_id: metadata.account_id }
```

Comment:
change this to:

```
data: { bank_account: { plaid_public_token: public_token, plaid_account_id: metadata.account_id } }
```

Reply:
Got it working by sticking that inside bank_account hash, as follows. :)
data: { bank_account: { plaid_public_token: public_token, plaid_account_id: metadata.account_id } }